proposals/testing/NNNN-targeted-interoperability-swift-testing-and-xctest.md

```diff
@@ -0,0 +1,292 @@
+
+- Opt-in **strict interop mode**, where XCTest API usage will result in
+  `fatalError("Usage of XCTest API in a Swift Testing context is not
+supported")`.
+
+Here are some concrete examples:
+
+| When running a Swift Testing test... | Current         | Proposed                                     | Proposed (strict) |
+| ------------------------------------ | --------------- | -------------------------------------------- | ----------------- |
+| `XCTAssert` failure is a ...         | ‼️ No-op        | ❌ Test Failure and ⚠️ Runtime Warning Issue | 💥 `fatalError`   |
+| `XCTAssert` success is a ...         | No-op           | ⚠️ Runtime Warning Issue                     | 💥 `fatalError`   |
+| `throw XCTSkip` is a ...             | ❌ Test Failure | ❌ Test Failure                              | ❌ Test Failure   |
+
+### Targeted support for Swift Testing APIs with XCTest API equivalents
+
+We propose supporting the following Swift Testing APIs in XCTest:
+
+- `#expect` and `#require`
+  - Includes [exit testing][]
+- `withKnownIssue`: marking an XCTest issue in this way will generate a runtime
+  warning issue. In strict interop mode, this becomes a `fatalError`.
+- Attachments
+- [Test cancellation][] (links to pitch)
+
+We think developers will find utility in using Swift Testing APIs in XCTest. For
+example, you can replace `XCTAssert` with `#expect` in your XCTest tests and
+immediately get the benefits of the newer, more ergonomic API. In the meantime,
+you can incrementally migrate the rest of your test infrastructure to use Swift
+Testing at your own pace.
+
+Present and future Swift Testing APIs will be supported in XCTest if the
+XCTest API _already_ provides similar functionality.
+
+- For example, we support the proposed Swift Testing [test cancellation][]
+  feature in XCTest since it is analogous to `XCTSkip`.
+
+- On the other hand, [Traits][] are a powerful Swift Testing feature, and
+  include the ability to [add tags][tags] to organise tests. Even though XCTest
+  does not interact with tags, **this is beyond the scope of interoperability**
+  because XCTest doesn't have existing “tag-like” behaviour to map onto.
+
+Here are some concrete examples:
+
+| When running a XCTest test...                | Current         | Proposed                 | Proposed (strict) |
+| -------------------------------------------- | --------------- | ------------------------ | ----------------- |
+| `#expect` failure is a ...                   | ‼️ No-op        | ❌ Test Failure          | ❌ Test Failure   |
+| `#expect` success is a ...                   | No-op           | No-op                    | No-op             |
+| `withKnownIssue` wrapping `XCTFail` is a ... | ❌ Test Failure | ⚠️ Runtime Warning Issue | 💥 `fatalError`   |
+
+### Interoperability Modes
+
+- **Warning-only**: This is for projects which do not want to see new test
+  failures surfaced due to interoperability.
```

Comment:
Can this bullet state what it _does_, i.e. that it records a warning-severity issue for test failure _and_ a second warning issue indicating the XCTest usage?